tests/lib/helpers.js

```diff
@@ -56,12 +56,15 @@ function startCasper(options) {
     var url = baseTestUrl + path;
 
     casper.echo('Starting with url: ' + url);
-    if (!headers) {
-        casper.start(url);
-    } else {
-        casper.start();
+    casper.start();
+    if (options.viewport) {
```

Comment:
I did this before, but I thought it was wiser to have the setUp/tearDown handle the viewport  because you have the viewPort changes propagate to other tests within the same file.

I made helpers desktopTest/tabletTest to wrap the test object to automatically inject viewport handlers